conda/core/python_dist.py

```diff
@@ -0,0 +1,444 @@
+      - https://packaging.python.org/specifications/core-metadata/
+    """
+
+    # TODO: Define precedence
+    METADATA_FILE_NAMES = ('METADATA', 'PKG-INFO', 'metadata.json')
+
+    def __init__(self, path):
+        """"""
+        self._path = path
+
+        metadata_path = self._process_metadata_path(path)
+        self._read_metadata(metadata_path)
+
+    def _process_metadata_path(self, path):
+        """"""
+        if isdir(path):
+            for fname in self.METADATA_FILE_NAMES:
+                fpath = join(path, fname)
+                if isfile(fpath):
+                    metadata_path = fpath
+                    break
+        elif isfile(path):
+            # '<pkg>.egg-info' file contains metadata directly
+            metadata_path = path
+        else:
+            raise MetadataConflictError
+
+        return metadata_path
+
+    def _read_metadata(self, fpath):
+        """"""
+        if isfile(fpath):
+            if fpath.endswith('.json'):
+                self._data = self._read_json_metadata(fpath)
+            else:
+                self._data = self._parse_metadata(fpath)
+        else:
+            self._data = {}
+
+    def _read_json_metadata(self, fpath):
+        """"""
+        data = None
+        if isfile(fpath):
+            with open(fpath, 'r') as f:
+                data = json.loads(f.read())
+        return data
+
+    def _parsed_metadata_to_json(self, data):
+        """Parse the original format which is stored as RFC-822 headers."""
+        new_data = OrderedDict() if data else None
+        if data:
+            for key, value in data.items():
+                new_key = key.lower().replace('-', '_')
+                if key in METADATA_MULTIPLE_KEYS:
+                    if new_key not in new_data:
+                        new_data[new_key] = [value]
+                    else:
+                        new_data[new_key].append(value)
+                elif key in METADATA_SINGLE_KEYS:
+                    new_data[new_key] = value
+                else:
+                    raise MetadataConflictError
+
+            # TODO: If we need to handle ALL edge cases we might just use the
+            # current metadata.py in distlib, or we could use a trimmed down
+            # version
+
+            # Multiple values should use plural keys?
+            # Handle body?
+            # Handle license?
+
+            # Handle keywords
+            if 'keywords' in new_data:
+                # TODO: Handle comma, semicolon?
+                new_data['keywords'] = new_data['keywords'].split()
+
+        return new_data
+
+    def _parse_metadata(self, fpath):
+        """"""
+        data = None
+        if isfile(fpath):
+            parser = email.parser.HeaderParser()
+            with open(fpath, 'r') as fp:
+                data = parser.parse(fp)
+        return self._parsed_metadata_to_json(data)
+
+    def get_run_requirements(self, exclude_markers=False):
+        """"""
```

Comment:
We are currently only using requires_dist, but there are more "interesting cases" any particular example of a "tricky" complex case to check for?

Reply:
We'll probably need to handle all or most of the edge cases.  Need to look for examples though.  Here's an interesting setup.py file: https://github.com/twilio/twilio-python/blob/master/setup.py

Reply:
Ok, will use this as an example ;-)